server/midleware/authenticateSpecificloan.js

```diff
@@ -1,25 +1,34 @@
 import jwt from 'jsonwebtoken';
-import { checkUserLoan } from '../helper/userHelper';
 import {
   validateToken, notValidToken, tokenError,
 } from '../helper/middlewareHelper';
+import pool from '../config/configDb';
+import { isOWnerQuery } from '../models/Queries';
 
 export default function checkToken(req, res, next) {
-  let token = req.headers['x-access-token'] || req.headers.authorization;
-  token = validateToken(token);
-  if (token) {
-    const { secret } = process.env;
-    jwt.verify(token, secret, (err, decoded) => {
-      if (err) {
-        return notValidToken(res);
-      }
-      req.decoded = decoded;
-      if (checkUserLoan(req.decoded.id, req.params.loanID) >= 0
-      || checkUserLoan(req.decoded.id, req.params.loanID) === -2) { next(); } else {
-        return tokenError(res);
-      }
-    });
+  const { loanID } = req.params;
+  if (loanID && !isNaN(loanID)) {
```

Comment:
`isNaN` is not a valid way to check if a value is an integer. For instance `isNaN(true)` returns false. You can use Number.isInteger(value)